app/javascript/App/components/NewsfeedPage/CoinListHeader.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import SectionHeader from './SectionHeader'
+import Switch from '../Switch'
+
+const CoinListHeader = (props) => {
+  const { toggleUI, currentUI } = props
+  return (
+    <SectionHeader>
+      <div className="flex items-center">
+        <Switch
+          on={currentUI('watchingOnly')}
+          onChange={() => toggleUI('watchingOnly')}
```

Comment:
any particular reason to return this invocation from an anonymous arrow function rather than just calling it?

Reply:
Yeah, wouldn't work otherwise right?
If toggleUI didn't have any arguments then we could indeed just do `onchange={toggleUI}`, but since we do then it needs to be in a function to avoid it being called on render

Reply:
well, you could use your good friend, bind 😄 

`toggleUI.bind(this, 'watchingOnly')`